Mathlib/Analysis/Normed/Unbundled/SpectralNorm.lean

```diff
@@ -644,4 +664,218 @@ theorem spectralAlgNorm_isPowMul : IsPowMul (spectralAlgNorm K L) := isPowMul_sp
+theorem spectralNorm_unique_field_norm_ext [CompleteSpace K]
+    {f : AbsoluteValue L ℝ} (hf_ext : ∀ (x : K), f (algebraMap K L x) = ‖x‖) (x : L) :
+    f x = spectralNorm K L x := by
+  set g : AlgebraNorm K L :=
+    { MulRingNorm.mulRingNormEquivAbsoluteValue.invFun f with
+      smul' := fun k x => by
+        simp only [AddGroupSeminorm.toFun_eq_coe, MulRingSeminorm.toFun_eq_coe]
+        rw [Algebra.smul_def, map_mul]
+        congr
+        rw [← hf_ext k]
+        rfl
+      mul_le' := fun x y => by
+        simp [AddGroupSeminorm.toFun_eq_coe, MulRingSeminorm.toFun_eq_coe] }
+  have hg_pow : IsPowMul g := MulRingNorm.isPowMul _
+  have hgx : f x = g x := rfl
+  rw [hgx, spectralNorm_unique hg_pow, spectralAlgNorm_def]
+
+/-- Given a nonzero `x : L`, the corresponding `seminormFromConst` can be regarded as an algebra
+  norm, when one assumes that `(spectralAlgNorm h_alg hna) 1 ≤ 1`. -/
+def algNormFromConst (h1 : (spectralAlgNorm K L).toRingSeminorm 1 ≤ 1) {x : L} (hx : x ≠ 0) :
+    AlgebraNorm K L :=
+  have hx' : spectralAlgNorm K L x ≠ 0 :=
+    ne_of_gt (spectralNorm_zero_lt hx (Algebra.IsAlgebraic.isAlgebraic x))
+  { normFromConst h1 hx' spectralAlgNorm_isPowMul with
+    smul' := fun k y => by
+      have h_mul : ∀ y : L, spectralNorm K L (algebraMap K L k * y) =
+          spectralNorm K L (algebraMap K L k) * spectralNorm K L y := fun y ↦ by
+        rw [spectralNorm_extends, ← Algebra.smul_def, ← spectralAlgNorm_def,
+          map_smul_eq_mul _ _ _, spectralAlgNorm_def]
+      have h : spectralNorm K L (algebraMap K L k) =
+        seminormFromConst' h1 hx' isPowMul_spectralNorm (algebraMap K L k) := by
+          rw [seminormFromConst_apply_of_isMul h1 hx' _ h_mul]; rfl
+      rw [← @spectralNorm_extends K _ L _ _ k, Algebra.smul_def, h]
+      exact seminormFromConst_isMul_of_isMul _ _ _ h_mul _ }
+
+theorem algNormFromConst_def (h1 : (spectralAlgNorm K L).toRingSeminorm 1 ≤ 1) {x y : L}
+    (hx : x ≠ 0) :
+    algNormFromConst h1 hx y =
+      seminormFromConst h1 (ne_of_gt (spectralNorm_zero_lt hx (Algebra.IsAlgebraic.isAlgebraic x)))
+        isPowMul_spectralNorm y := rfl
+
+/-- If `K` is a field complete with respect to a nontrivial nonarchimedean multiplicative norm and
+  `L/K` is an algebraic extension, then the spectral norm on `L` is multiplicative. -/
+theorem spectralAlgNorm_mul [CompleteSpace K] (x y : L) :
+    spectralAlgNorm K L (x * y) = spectralAlgNorm K L x * spectralAlgNorm K L y := by
+  by_cases hx : x = 0
+  · simp [hx, zero_mul, map_zero]
+  · have hx' : spectralAlgNorm K L x ≠ 0 :=
+      ne_of_gt (spectralNorm_zero_lt hx (Algebra.IsAlgebraic.isAlgebraic x))
+    have hf1 : (spectralAlgNorm K L) 1 ≤ 1 := le_of_eq spectralAlgNorm_one
+    set f : AlgebraNorm K L := algNormFromConst hf1 hx with hf
+    have hf_pow : IsPowMul f := seminormFromConst_isPowMul hf1 hx' isPowMul_spectralNorm
+    rw [← spectralNorm_unique hf_pow, hf]
+    simp [algNormFromConst_def]
```

Comment:
This is a non-terminal simp, right? Shouldn't a linter be complaining about this? Maybe squeeze it?

Reply:
It seems I can just remove it.